components/input/navigationRectangle.js

```diff
@@ -1,130 +1,100 @@
-        </Text>
-      );
-    }
-    //Club Requirement takes next priority in multi-requirement Drills
-    if (requirementName == "club") {
-      return (
-        <Text
-          style={{
-            fontWeight: "bold",
-            fontSize: 16,
-            padding: 2,
-          }}
-        >
-          {drillInfo.requirements[0].prompt}:{" "}
-          {attemptShots[shotIndex - 1].items.club}
-        </Text>
-      );
-    }
-  }
-
-  //Return empty if no requirement display is found
-  console.log(
-    "Requirement Display error in requirementDisplay located within navitgationRectangle component",
-  );
-  return <></>;
-}
-
-export default function NavigationRectangle({
-  drillInfo,
-  attemptShots,
-  inputValues,
-  shotIndex,
-}) {
+export default function NavigationRectangle({ drillInfo, inputValues, shot }) {
+  console.log("inputValues", inputValues);
+  const keys = Object.keys(inputValues);
   return (
     <View
       style={{
         backgroundColor: "#d9d9d9",
-        padding: 20,
+        padding: 10,
+        paddingLeft: 20,
+        paddingRight: 20,
         borderRadius: 10,
         maxHeight: 250,
         width: "80%",
         overflow: "hidden",
-        marginBottom: 20,
         flexDirection: "row",
         alignItems: "center",
         justifyContent: "space-between",
-        elevation: 5,
       }}
     >
-      <View style={{ flex: 1 }}>
+      <View style={{}}>
         <Text
           style={{
-            fontWeight: "bold",
-            fontSize: 25,
-            elevation: 5,
-            textShadowColor: "rgba(0, 0, 0, 0.2)",
-            textShadowOffset: { width: 2, height: 2 },
-            textShadowRadius: 5,
+            fontSize: 20,
           }}
         >
-          Shot {shotIndex}/{attemptShots.length}
+          Shot <Text style={{ fontWeight: "bold" }}>{shot.shotNum}</Text>
         </Text>
       </View>
 
       <View
         style={{
-          flex: 1,
           flexDirection: "column",
           alignItems: "center",
-          justifyContent: "space-between",
+          justifyContent: "center",
         }}
       >
         <View
           style={{
-            marginBottom: 10,
-            borderBottomWidth: 2,
-            borderColor: "#A0A0A0",
+            flexDirection: "row",
+            justifyContent: "space-between",
+            gap: 10,
           }}
         >
-          {requirementDisplay(attemptShots, drillInfo, shotIndex)}
+          {drillInfo.requirements.map((requirement) => (
+            <Text
+              key={requirement.name}
+              style={{
+                fontSize: 16,
+                padding: 2,
+              }}
+            >
+              {requirement.prompt}:
+              <Text style={{ fontWeight: "bold" }}>
+                {" "}
+                {shot.items[requirement.name]} {requirement.distanceMeasure}
+              </Text>
+            </Text>
+          ))}
         </View>
```

Comment:
More of a concern than a change, the original component was only displaying one of the requirements to avoid cutter. My concern comes if there is a drill in the future that has 3+ requirements. This would certainly be cluttered and have some display issues but I guess we can cross that bridge when we get there if we need to handle those drills.

Reply:
Additionally, for the full line test, there is a weird display error with the distance measure. It might just be how it is implemented on the database. 
![image](https://github.com/Golf-Drill-Challenge-App/Golf-App/assets/91503842/5201199d-377a-4627-bcd1-d323042f2b3b)

Reply:
It probably is. As for the 3+ requirement situation, we can either remove requirements altogether or limit it to 1 or 2 like what you've done.

Reply:
I've fixed the weird display thing. It indeed was a database error